app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -533,6 +575,19 @@ open class ProfilesViewModel(
         onFailure = { e -> handleError(e) })
   }
 
+  /** Fetches the current user's profile from the repository. */
+  fun getSelfProfileAndThen(andThen: () -> Unit) {
+    _loadingSelf.value = true
+    repository.getProfileByUid(
+        auth.currentUser?.uid ?: "null",
+        onSuccess = { profile ->
+          _selfProfile.value = profile
+          _loadingSelf.value = false
+          andThen()
+        },
+        onFailure = { e -> handleError(e) })
+  }
```

Comment:
I can make that too because `getSelfProfileAndThen` is the race-proof version of `getSelfProfile`, so refactoring it into one that has an `andThen` can be good for code duplication, race conditions, scalability and maintainability.